spec/catalog/api/minion/task_spec.rb

```diff
@@ -1,50 +1,50 @@
 require "catalog/minion"
 
-describe Catalog::Api::Minion::Order do
+describe Catalog::Api::Minion::Task do
   describe "#perform" do
-    let(:order) do
+    let(:task_minion) do
       described_class.new("localhost", "9092")
     end
 
     let(:message) { ManageIQ::Messaging::ReceivedMessage.new(nil, "Task.update", payload, nil, nil, nil) }
 
     let(:payload) { {"task_id" => "123"} }
     let(:payload_params) do
-      {:payload =>  message.payload, :message => message.message}
+      {:payload => message.payload, :message => message.message}
     end
     let(:config) do
       test_config = ::CatalogApiClient.configure
-      test_config.host = "catalog-api.catalog-ci.svc:8080"
+      test_config.host = "localhost:3000"
     end
 
     before do
       dummy_client = double("CatalogApiClient")
       allow(dummy_client).to receive(:configure).and_return(config)
-      stub_request(:post, "http://catalog-api.catalog-ci.svc:8080/internal/v1.0/notify/order_item/123")
+      stub_request(:post, "http://localhost:3000/internal/v1.0/notify/task/123")
     end
 
     context "when there is no error" do
-      it "posts a payload" do
-        order.perform(message)
-        expect(a_request(:post, "http://catalog-api.catalog-ci.svc:8080/internal/v1.0/notify/order_item/123").with(
+      it "posts a payload to the order item endpoint" do
+        task_minion.perform(message)
+        expect(a_request(:post, "http://localhost:3000/internal/v1.0/notify/task/123").with(
           :body    => payload_params,
           :headers => {
-            'X-Rh-Identity'=>'eyJlbnRpdGxlbWVudHMiOnsiaHlicmlkX2Nsb3VkIjp7ImlzX2VudGl0bGVkIjp0cnVlfX0sImlkZW50aXR5Ijp7ImFjY291bnRfbnVtYmVyIjoiY2F0YWxvZy1hcGktb3JkZXItbWluaW9uIn19'
+            'X-Rh-Identity'=>'eyJlbnRpdGxlbWVudHMiOnsiaHlicmlkX2Nsb3VkIjp7ImlzX2VudGl0bGVkIjp0cnVlfX0sImlkZW50aXR5Ijp7ImFjY291bnRfbnVtYmVyIjoiY2F0YWxvZy1hcGktdGFzay1taW5pb24ifX0='
```

Comment:
Interesting, why did the header need to change?

Reply:
Good question, I'm not sure. The spec kept failing and so I ended up having to copy the header to fix it. The header in the `approval_spec.rb` is the old one still and it still works, so yeah, I'm not sure.

Reply:
it seems the account_number is changing to match the name of the minion.
```
irb(main):005:0> Base64.decode64('eyJlbnRpdGxlbWVudHMiOnsiaHlicmlkX2Nsb3VkIjp7ImlzX2VudGl0bGVkIjp0cnVlfX0sImlkZW50aXR5Ijp7ImFjY291bnRfbnVtYmVyIjoiY2F0YWxvZy1hcGktb3JkZXItbWluaW9uIn19')
=> "{\"entitlements\":{\"hybrid_cloud\":{\"is_entitled\":true}},\"identity\":{\"account_number\":\"catalog-api-order-minion\"}}"
irb(main):006:0> new_str = Base64.decode64('eyJlbnRpdGxlbWVudHMiOnsiaHlicmlkX2Nsb3VkIjp7ImlzX2VudGl0bGVkIjp0cnVlfX0sImlkZW50aXR5Ijp7ImFjY291bnRfbnVtYmVyIjoiY2F0YWxvZy1hcGktdGFzay1taW5pb24ifX0')
=> "{\"entitlements\":{\"hybrid_cloud\":{\"is_entitled\":true}},\"identity\":{\"account_number\":\"catalog-api-task-minion\"}}"
```

We should probably document this so there is a better understanding of why a specific header is used.

Reply:
Well, that's weird, because the spec for the `catalog-api-approval-minion` has that identity header and it passes.